test/kafka-sink-stalled-errors/stalled.td

```diff
@@ -0,0 +1,42 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ kafka-create-topic topic=status-history
+
+> CREATE CONNECTION kafka_conn
+  TO KAFKA (BROKER '${testdrive.kafka-addr}');
+
+> CREATE CONNECTION IF NOT EXISTS csr_conn TO CONFLUENT SCHEMA REGISTRY (
+    URL '${testdrive.schema-registry-url}'
+  );
+
+> CREATE TABLE sink_table (f1 INTEGER);
+
+> CREATE SINK kafka_sink FROM sink_table
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-kafka-sink-${testdrive.seed}')
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+  ENVELOPE DEBEZIUM
+
+$ set-from-sql var=sink_id
+SELECT id FROM mz_sinks WHERE name = 'kafka_sink'
+
+$ set-regex match="\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d(\.\d\d\d)?" replacement="<TIMESTAMP>"
+
+# Verify we get a starting status
+> SELECT * FROM mz_internal.mz_sink_status_history WHERE sink_id = '${sink_id}' ORDER BY occurred_at LIMIT 1;
+"<TIMESTAMP> UTC" ${sink_id} starting <null> <null>
+
+> INSERT INTO sink_table VALUES (1);
+
+> INSERT INTO sink_table VALUES (2);
+
+# We should get a stalled error message next
+> SELECT status, sink_id, error, details::json#>>'{hint}' as hint FROM mz_internal.mz_sink_status_history
```

Comment:
Instead of the `details::json#>>'{hint}'`, I was trying to do `jsonb_extract_path_text(details, 'hint')` from [link](https://www.postgresql.org/docs/9.5/functions-json.html), but it kept giving me error
```
ERROR:  function jsonb_extract_path_text(jsonb, unknown) does not exist
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

Reply:
We don't support `jsonb_extract_path_text`; you can find the builtins in `src/sql/src/func.rs`

Reply:
Oh I see, thanks for the pointer 🙏 !